packages/eslint-plugin/src/rules/explicit-member-accessibility.ts

```diff
@@ -15,30 +37,132 @@ export default util.createRule({
     messages: {
       missingAccessibility:
         'Missing accessibility modifier on {{type}} {{name}}.',
+      unwantedPublicAccessibility:
+        'Public accessibility modifier on {{type}} {{name}}.',
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          accessibility: accessibilityLevel,
+          overrides: {
+            type: 'object',
+            properties: {
+              accessors: accessibilityLevel,
+              constructors: accessibilityLevel,
+              methods: accessibilityLevel,
+              properties: accessibilityLevel,
+              parameterProperties: accessibilityLevel,
+            },
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
   },
-  defaultOptions: [],
-  create(context) {
+  defaultOptions: [{ accessibility: 'explicit' }],
+  create(context, [option]) {
+    /**
+     * Reads the value set on the Override and returns a Check value
+     * Check value is used to control what, if any accessibility modifiers are required or banned
+     */
+    function parseOverride(
+      defaultCheck: AccessibilityLevel,
+      overrideToCheck?: AccessibilityLevel,
+    ): AccessibilityLevel {
+      return typeof overrideToCheck === 'undefined'
+        ? defaultCheck
+        : overrideToCheck;
+    }
+
+    let baseCheck: AccessibilityLevel = 'explicit';
+    if (option.accessibility) {
+      baseCheck = option.accessibility;
+    }
+    let ctorCheck = baseCheck;
+    let accessorCheck = baseCheck;
+    let methodCheck = baseCheck;
+    let propCheck = baseCheck;
+    let paramPropCheck = baseCheck;
+    if (option.overrides) {
+      ctorCheck = parseOverride(baseCheck, option.overrides.constructors);
+      accessorCheck = parseOverride(baseCheck, option.overrides.accessors);
+      methodCheck = parseOverride(baseCheck, option.overrides.methods);
+      propCheck = parseOverride(baseCheck, option.overrides.properties);
+      paramPropCheck = parseOverride(
+        baseCheck,
+        option.overrides.parameterProperties,
+      );
+    }
+
+    /**
+     * Generates the report for rule violations
+     */
+    function reportIssue(
+      messageId: MessageIds,
+      nodeType: string,
+      node:
+        | TSESTree.MethodDefinition
```

Comment:
Just to make this definition simpler, because you don't access the properties of the node, you can just type it as:

```TS
node: TSESTree.Node`
```

That way if you add more types later it won't force you to add extra lines.

Reply:
done